app/controllers/application_controller.rb

```diff
@@ -1,2 +1,13 @@
-class ApplicationController < ActionController::Base
-end
+class ApplicationController < ActionController::Base
+  protect_from_forgery prepend: true
+  before_action :authenticate_user!

```

Comment:
I think it would be better not to implement authentication to public recipes. CR's can reject it because its not possible to access public recipes currently.